File: frontend/src/app/dto/quizGameRound.ts
import {Player} from './player';
import {QuizAnswer} from './quizAnswer';
import {GameRoom} from './gameRoom';

export class QuizGameRound {
  id: number;
  round: number;
  player: Player;
  gameRoom: GameRoom;
  answer1: QuizAnswer;
  answer2: QuizAnswer;
  answer3: QuizAnswer;
  answer4: QuizAnswer;
  answer5: QuizAnswer;
  answer6: QuizAnswer;
  answer7: QuizAnswer;
  answer8: QuizAnswer;
  answer9: QuizAnswer;
  answer10: QuizAnswer;
  answerTheme: QuizAnswer;
  type: string;

  constructor(args: QuizGameRound = <QuizGameRound>{}) {
    this.id = args.id;
    this.round = args.round;
    this.player = args.player;
    this.gameRoom = args.gameRoom;
    this.answer1 = args.answer1;
    this.answer2 = args.answer2;
    this.answer3 = args.answer3;
    this.answer4 = args.answer4;
    this.answer5 = args.answer5;
    this.answer6 = args.answer6;
    this.answer7 = args.answer7;
    this.answer8 = args.answer8;
    this.answer9 = args.answer9;
    this.answer10 = args.answer10;
    this.answerTheme = args.answerTheme;
    this.type = args.type;
  }

}

File: src/main/java/com/corelogic/schemaconverter/games/quiz/entity/QuizGameRound.java
package com.corelogic.schemaconverter.games.quiz.entity;

import com.corelogic.schemaconverter.entity.GameRoom;
import com.corelogic.schemaconverter.entity.Player;
import com.corelogic.schemaconverter.entity.enums.GameRoomType;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class QuizGameRound {

    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "QuizGameRoundSeq")
    @SequenceGenerator(name = "QuizGameRoundSeq", sequenceName = "quiz_game_round_id_seq", allocationSize = 1)
    @Id
    private Long id;

    private Long round;

    @OneToOne
    private GameRoom gameRoom;

    @OneToOne
    private Player player;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer1;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer2;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer3;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer4;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer5;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer6;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer7;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer8;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer9;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answer10;

    @ManyToOne(cascade = CascadeType.ALL)
    private QuizAnswer answerTheme;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    @Transient
    private GameRoomType type = GameRoomType.QUIZ;
}

